lms/djangoapps/dashboard/sysadmin.py

```diff
@@ -570,6 +571,28 @@ def post(self, request):
         return self.get(request)
 
 
+def get_course(course_id):
+    """
+    returns course object, checks if split module store
+    is set then uses CourseLocator to fin course.
+    """
+    try:
+        return get_course_by_id(course_id)
```

Comment:
Is `course_id` supposed to be a string or a `CourseKey` object?

If you're parsing from a string, doing `CourseKey.from_string(course_id)` should return you an appropriate `CourseKey` sub-type, which should route to the correct modulestore when you `get_course_by_id`

Reply:
if you are in split module store and use slash course key for  input of `CourseKey.from_string(course_id)` it wont parse it in format i.e `course-v1:FAST+CS102+2018_FEB_1`. This cases issue of 404 here. So in my code i check if course is not available then i reformat key into format `course-v1:ORG+CS102+COURSE` and it worked fine. tests are also added

Reply:
Wait, just so I'm clear on this... the input to this function can sometimes be a slash separated, old-style course key, even when it's talking about data that is in the split modulestore? Why do we allow people to specify a course identifier that isn't the real course key?